### PR TITLE
[chore][mdatagen] skip generation of testdata directory if no metrics are defined

### PR DESCRIPTION
**Pull request by atoulme**
Do not generate testdata folder if no metrics are present.
Add tests for the generation.

### CODE REVIEW

File: .chloggen/loki-exporter-add-raw-log-export-format.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+change_type: enhancement
+component: translator/loki
+note: Loki add raw log export format
+issues: [ 18888 ]
```
Comment (dmitryax) [inline .chloggen/loki-exporter-add-raw-log-export-format.yaml:4]: Rebasing issue?

**Comment by atoulme** [comment]
Rebase on top of #19772

File: .chloggen/loki-exporter-add-raw-log-export-format.yaml (hunk 1 above)
Comment (atoulme) [reply .chloggen/loki-exporter-add-raw-log-export-format.yaml:4]: #19772 should be merged first, sorry. I rebased on top of it to add tests and a few more things to ignore.

**Comment by atoulme** [comment]
@codeboten you might be best placed to review those changes.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**